src/Umbraco.Infrastructure/Routing/RedirectTrackingHandler.cs

```diff
@@ -121,10 +145,26 @@ private void StoreOldRoute(IContent entity, OldRoutesDictionary oldRoutes)
                     var route = contentCache.GetRouteById(publishedContent.Id, culture);
                     if (IsNotRoute(route))
```

Comment:
Just a minor stylistic thing, but I'd reverse the if-statement and use and else-if 

```c#
var route = contentCache.GetRouteById(publishedContent.Id, culture);
if (!IsNotRoute(route))
{
    oldRoutes[new ContentIdAndCulture(publishedContent.Id, culture)] = new ContentKeyAndOldRoute(publishedContent.Key, route);
}
else if (string.IsNullOrEmpty(culture))
{
    // Retry using all languages, if this is invariant but has a variant ancestor
    var languages = _localizationService.GetAllLanguages();
    foreach (var language in languages)
    {
        route = contentCache.GetRouteById(publishedContent.Id, language.IsoCode);
        if (!IsNotRoute(route))
        {
            oldRoutes[new ContentIdAndCulture(publishedContent.Id, language.IsoCode)] = new ContentKeyAndOldRoute(publishedContent.Key, route);
        }
    }
}
```